File: llm_generated_code/ChatGPT/CPP/Level1/Q2.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

int minOperations(vector<int>& a) {
    int oddCount = 0, evenCount = 0;
    for (int num : a) {
        if (num % 2 == 0) {
            evenCount++;
        } else {
            oddCount++;
        }
    }
    // The minimum number of operations needed to make all elements have the same parity
    // is the minimum of converting all to even or all to odd
    return min(oddCount, evenCount);
}

int main() {
    int t;
    cin >> t;
    while (t--) {
        int n;
        cin >> n;
        vector<int> a(n);
        for (int i = 0; i < n; ++i) {
            cin >> a[i];
        }
        cout << minOperations(a) << endl;
    }
    return 0;
}

File: llm_generated_code/ChatGPT/Python/Level1/Q2.py
def min_operations(arr):
    odd_count = sum(1 for x in arr if x % 2 != 0)
    even_count = len(arr) - odd_count
    return min(odd_count, even_count)

def main():
    t = int(input())
    results = []
    for _ in range(t):
        n = int(input())
        arr = list(map(int, input().split()))
        results.append(min_operations(arr))
    
    for result in results:
        print(result)

if __name__ == "__main__":
    main()
